catroid/src/org/catrobat/catroid/utils/StatusBarNotificationManager.java

```diff
@@ -166,8 +179,152 @@ public void abortProgressNotificationWithMessage(int id, String changeDoneText)
 		showOrUpdateNotification(id, MAXIMUM_PERCENT);
 	}
 
+	public void showUploadRejectedNotification(int id, int statusCode, String serverAnswer, Bundle bundle) {
+		NotificationData notificationData = notificationDataMap.get(id);
+
+		if (notificationData == null) {
+			Log.d(TAG, "NotificationData is null.");
+			return;
+		}
+		NotificationCompat.Builder notificationBuilder = notificationData.getNotificationBuilder();
+		notificationData.setNotificationTextDone(serverAnswer);
+
+		Uri alarmSound = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
+		notificationBuilder.setContentTitle(context.getResources().getText(R.string.notification_upload_rejected))
+				.setContentText(serverAnswer)
+				.setTicker(context.getResources().getText(R.string.notification_upload_rejected))
+				.setSound(alarmSound)
+				.setStyle(new NotificationCompat.BigTextStyle().bigText(serverAnswer))
+				.setProgress(0, 0, false)
+				.setAutoCancel(true)
+				.setPriority(Notification.PRIORITY_MAX)
+				.setOngoing(false);
+
+		switch (statusCode) {
+			case 500:
+			case 501:
+			case 504:
+			case 505:
+			case 506:
+			case 507:
+			case 513:
+			case 514:
```

Comment:
Maybe there are status codes in the android library? Otherwise you have to define those nasty magic numbers yourself.